### PR TITLE
Update form.css

### PR DESCRIPTION
**Pull request by Floxn**
Issue from [this](https://github.com/contao/core/issues/8747)

### CODE REVIEW

**Comment by leofeyer** [comment]
Hm, none of the other form elements is prefixed with `form` though. Maybe we should prefix them all?

**Comment by leofeyer** [comment]
@aschempp @ausi @frontendschlampe Are you ok with the changes?

**Comment by aschempp** [comment]
Can't tell, I never use that file.

File: css/form.css [hunk 1]
```diff
@@ -187,29 +187,29 @@ button.red,input[type="submit"].red,.button.red {
...
-.widget > input,.widget > textarea {
+form .widget > input,form .widget > textarea {
 	width:100%;
 }
 
 /**
  * Inline form layout
  */
-.inline-form .widget {
+form .inline-form .widget {
```
Comment (leofeyer) [inline css/form.css]: This is wrong, because `.inline-form` is on the `<form>` element itself.

File: css/form.css [hunk 2]
```diff
@@ -218,21 +218,21 @@ button.red:active,input[type="submit"].red:active,.button.red:active {
 /**
  * Horizontal form layout
  */
-.horizontal-form .widget:after {
+form .horizontal-form .widget:after {
```
Comment (leofeyer) [inline css/form.css]: This is wrong, because `.horizontal-form` is on the `<form>` element itself.

File: css/form.css [hunk 3]
```diff
@@ -218,21 +218,21 @@ button.red:active,input[type="submit"].red:active,.button.red:active {
...
 	content:"";
 	display:table;
 	clear:both;
 }
-.horizontal-form .widget > label {
+form .horizontal-form .widget > label {
```
Comment (leofeyer) [inline css/form.css]: This is wrong, because `.horizontal-form` is on the `<form>` element itself.

File: css/form.css [hunk 4]
```diff
@@ -218,21 +218,21 @@ button.red:active,input[type="submit"].red:active,.button.red:active {
...
 	width:20%;
 	float:left;
 	padding:3px 2% 0 0;
 	text-align:right;
 }
-.horizontal-form .widget > input,.horizontal-form .widget > textarea {
+form .horizontal-form .widget > input,form .horizontal-form .widget > textarea {
```
Comment (leofeyer) [inline css/form.css]: This is wrong, because `.horizontal-form` is on the `<form>` element itself.

File: css/form.css [hunk 5]
```diff
@@ -218,21 +218,21 @@ button.red:active,input[type="submit"].red:active,.button.red:active {
...
 	width:78%;
 	float:left;
 }
-.horizontal-form .widget > fieldset {
+form .horizontal-form .widget > fieldset {
```
Comment (leofeyer) [inline css/form.css]: This is wrong, because `.horizontal-form` is on the `<form>` element itself.

**Comment by leofeyer** [comment]
I think the main purpose of the `.button` class is to have `<a class="button">` elements that are styled the same as the `<button>` elements. Not sure if we have this case though.

**Comment by Floxn** [comment]
i´ve fixed it

**Comment by leofeyer** [comment]
@fritzmg @frontendschlampe What do you think about this change?

**Comment by fritzmg** [comment]
imho the general selector should stay `form`, not `.formbody`. If you use the Contao CSS Framework (and the form styles specifically) and you include a form into your site that is not generated by Contao, then you might want to have this form look the same way as the other forms as well.

However, I do agree with @ausi about the problem with `.button` (`.widget` is imho too Contao specific).

**Comment by leofeyer** [comment]
I have adjusted the PR according to your suggestions.

@Floxn Can you please check if the changes fix your issue?

@ausi @fritzmg Can you please review and approve?

**Comment by fritzmg** [comment]
Looks good to me 👍

**Comment by leofeyer** [comment]
But only prefixing `button` is not correct either, is it? Then every form element outside a form would get a default styling except buttons.

**Comment by ausi** [comment]
> Then every form element outside a form would get a default styling except buttons.

And that should be enough to fix contao/core#8747 IMO.

> But only prefixing `button` is not correct either, is it?

I think there is no *“correct”* answer, it depends on what the scope of the *form.css* should be. But because I never used that CSS file, I might be the wrong person to tell.

**Comment by fritzmg** [comment]
If we _only_ want to fix https://github.com/contao/core/issues/8747 couldn't we use something like
```css
button:not(#cboxNext):not(#cboxPrev)
```

**Comment by ausi** [comment]
`button:not(#cboxNext):not(#cboxPrev)` has a very high specificity. This would make it hard to overwrite button styles.

**Comment by leofeyer** [comment]
> The current state of this pull request would be a big BC break IMO, because it increases the specificity of most form selectors.

You are right. We must only prefix the `button` selector.

**Pull request merged**